h2o-core/src/main/java/water/rapids/ast/prims/string/AstReplaceAll.java

```diff
@@ -65,16 +66,29 @@ public Val apply(Env env, Env.StackHelp stk, AstRoot asts[]) {
   }
 
   private Vec replaceAllCategoricalCol(Vec vec, String pattern, String replacement, boolean ignoreCase) {
+    final Pattern compiledPattern = Pattern.compile(pattern); // Compile the pattern once before replacement
```

Comment:
This is actually a speedup, as the previous version had to effectively call Pattern.compile every time with String.replaceAll. This change is present four times - once for string vectors and once for categorical vectors branch in both AstReplaceAll and AstReplaceFirst - four times totally.